Tweak.xm

```diff
@@ -1,69 +1,38 @@
-    NSDate* date = [NSDate date];
-    NSCalendar *gregorian = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
-    NSDateComponents *hourComp = [gregorian components:NSCalendarUnitHour fromDate:date];
-    NSDateComponents *minuteComp = [gregorian components:NSCalendarUnitMinute fromDate:date];
-
-    //this gives us a 24hr formatted time, so convert it to 12hr format for ease of programming
-    int formattedHour = ([hourComp hour] >= 12) ? [hourComp hour] - 12 : [hourComp hour];
-    int formattedMinute = [minuteComp minute];
-
-    NSLog(@"Hour: %i", formattedHour);
-    NSLog(@"Minute: %i", formattedMinute);
 
-    if (formattedHour == 4 && formattedMinute == 20) {
-        //smoke day every weed
+-(void)_updateView {
 
-        UIImageView* snoop = [[UIImageView alloc] initWithImage:[UIImage imageWithContentsOfFile:@"/Library/Application Support/TokeTime/snoop.png"]];
-        snoop.frame = CGRectMake(-snoop.frame.size.width/4, snoop.frame.size.height/8, snoop.frame.size.width*0.9, snoop.frame.size.height*0.9);
-        snoop.alpha = 0.0;
-        snoop.tag = kSnoopTag;
-        //add view at the back of the LS view
-        [_lsView insertSubview:snoop atIndex:0];
-        [UIView animateWithDuration:1.0 animations:^{
-            snoop.alpha = 1.0;
-        }];
-    }
-    else {
-        //if we've added snoop to the view
-        UIImageView* snoop = (UIImageView*)[_lsView viewWithTag:kSnoopTag];
-        if (snoop != nil) {
-            [UIView animateWithDuration:1.0 animations:^{
-                snoop.alpha = 0.0;
-            } completion:^(BOOL finished){
-                if (finished) [snoop removeFromSuperview];
-            }];
+    %orig;
+    NSDate * now = [NSDate date];
+    NSDateFormatter *outputFormatter = [[NSDateFormatter alloc] init];
+    [outputFormatter setDateFormat:@"HH:mm"];
+    NSString *newDateString = [outputFormatter stringFromDate:now];
+    UIView *v = [self view];
+    
+    // you can also convert them
+    if([[NSString stringWithFormat: @"%@", newDateString] isEqualToString: @"04:20"] || [[NSString stringWithFormat: @"%@", newDateString] isEqualToString: @"4:20"] || [[NSString stringWithFormat: @"%@", newDateString] isEqualToString: @"16:20"]) {
+      UIImageView *snoopImageView = [[UIImageView alloc] initWithImage:[UIImage imageWithContentsOfFile:@"/Library/Application Support/HighDate/s.jpg"]];
+      snoopImageView.frame = v.superview.bounds;
+      snoopImageView.tag = 69;
+      snoopImageView.contentMode = UIViewContentModeScaleAspectFill;
+
+      if (![snoopImageView isDescendantOfView: v.superview.superview.superview]) {
+        [v.superview.superview.superview insertSubview: snoopImageView atIndex: 0];
+      }
+    } else {
+        // directly getting the view with the tag id doess not seem to work, it works for now, i'll look into this later
+        for(UIView *view in v.superview.superview.superview.subviews) {
+            if(view.tag == 69) {
```

Comment:
Can we make the tag here less likely to collide with another view?

```suggestion
#define kOverlayViewTag someUniqueNumber
            if(view.tag == kOverlayViewTag) {
```

Reply:
Yeah sorry forgot to change it